File: java/src/main/java/deltix/luminary/Int64Def.java
package deltix.luminary;

import java.util.ArrayList;
import java.util.List;

public class Int64Def implements TypeDef<TypeInt64>, ConstantScope {
    private List<ConstantDef> definedConstants;

    private Int64Def() {
        definedConstants = new ArrayList<>();
        definedConstants.add(new ConstantDef(this, "MIN_VALUE", TypeInt64.INSTANCE, new LiteralInt64(Long.MIN_VALUE)));
        definedConstants.add(new ConstantDef(this, "MAX_VALUE", TypeInt64.INSTANCE, new LiteralInt64(Long.MAX_VALUE)));
    }

    public static final Int64Def INSTANCE = new Int64Def();

    @Override
    public String getName() {
        return "UInt64";
    }

    @Override
    public String getFullName() {
        return getName();
    }

    @Override
    public List<ConstantDef> getDefinedConstants() {
        return definedConstants;
    }

    @Override
    public TypeInt64 getType() {
        return TypeInt64.INSTANCE;
    }

    @Override
    public FileDef getFile() {
        return null;
    }

    @Override
    public TypeScope getParent() {
        return null;
    }
}

File: csharp/main/Int64Def.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	public class Int64Def : IConstantScope, ITypeDef<TypeInt64>
	{
		private Int64Def()
		{
			DefinedConstants.Add(new ConstantDef(this, "MIN_VALUE", TypeInt64.Instance, new LiteralInt64(Int64.MinValue)));
			DefinedConstants.Add(new ConstantDef(this, "MAX_VALUE", TypeInt64.Instance, new LiteralInt64(Int64.MaxValue)));
		}

		public static Int64Def Instance = new Int64Def();

		public String Name => "Int64";

		public String FullName => Name;

		public List<ConstantDef> DefinedConstants { get; } = new List<ConstantDef>();

		public TypeInt64 Type => TypeInt64.Instance;

		public FileDef File => null;

		public ITypeScope Parent => null;
	}
}
